crates/nu-utils/standard_library/tests.nu

```diff
@@ -1,135 +1,19 @@
-use std.nu
-
-def test_assert [] {
-    def test_failing [code: closure] {
-        let code_did_run = (try { do $code; true } catch { false })
-
-        if $code_did_run {
-            error make {msg: (view source $code)}
-        }
-    }
-
-    std assert true
-    std assert (1 + 2 == 3)
-    test_failing { std assert false }
-    test_failing { std assert (1 + 2 == 4) }
-
-    std assert eq (1 + 2) 3
-    test_failing { std assert eq 1 "foo" }
-    test_failing { std assert eq (1 + 2) 4) }
-
-    std assert ne (1 + 2) 4
-    test_failing { std assert ne 1 "foo" }
-    test_failing { std assert ne (1 + 2) 3) }
-}
-
-def tests [] {
-    use std.nu assert
-
-    let branches = {
-        1: { -1 }
-        2: { -2 }
-    }
-
-    assert ((std match 1 $branches) == -1)
-    assert ((std match 2 $branches) == -2)
-    assert ((std match 3 $branches) == $nothing)
-
-    assert ((std match 1 $branches { 0 }) == -1)
-    assert ((std match 2 $branches { 0 }) == -2)
-    assert ((std match 3 $branches { 0 }) == 0)
-}
-
-def test_path_add [] {
-    use std.nu "assert eq"
-
-    with-env [PATH []] {
-        assert eq $env.PATH []
-
-        std path add "/foo/"
-        assert eq $env.PATH ["/foo/"]
-
-        std path add "/bar/" "/baz/"
-        assert eq $env.PATH ["/bar/", "/baz/", "/foo/"]
-
-        let-env PATH = []
-
-        std path add "foo"
-        std path add "bar" "baz" --append
-        assert eq $env.PATH ["foo", "bar", "baz"]
-
-        assert eq (std path add "fooooo" --ret) ["fooooo", "foo", "bar", "baz"]
-        assert eq $env.PATH ["fooooo", "foo", "bar", "baz"]
-    }
-}
-
-
-def test_dirs [] {
-
-    def "myassert" [
-        predicate: bool
-        msg?:string = "..."
-        --verbose = false (-v)  # enable to see successful tests
-    ] {
-        if not $predicate {
-            let span = (metadata $predicate).span
-            error make {msg: $"Assertion failed checking ($msg)",
-                        label: {text: "Condition not true" start: $span.start end: $span.end}}
-        } else {
-            if $verbose {
-                echo $"check succeeded: ($msg)"
-            }
+def main [] {
+    for test_file in (ls ($env.FILE_PWD | path join "test_*.nu") -f | get name) {
+        let $module_name = ($test_file | path parse).stem
+
+        echo $"(ansi yellow)INFO Run tests in ($module_name)(ansi reset)"
+        let tests = (
+            nu -c $'use ($test_file) *; $nu.scope.commands | to nuon'
+            | from nuon
+            | where module_name == $module_name
+            | where ($it.name | str starts-with "test_")
+            | get name
+        )
+
+        for test_case in $tests {
+            echo $"(ansi grey)DEBUG Run test ($module_name)/($test_case)(ansi reset)"
```

Comment:
i only have a small concern about the `grey`.

> **Warning**
> this might have to do with the color scheme **however** i'm using the default dark one, so that is probably a problem in the end

on the top with `grey`
on the bottom with `default_dimmed`
![colors](https://user-images.githubusercontent.com/44101798/225394549-e52536f9-480f-4bcb-b4a4-0a1b280ede55.png)

Reply:
Good idea, I changed it. Please note that I consider these code lines temporary until #8448 is merged.

Reply:
yup makes sense :+1: 

i think the default colors are just fine for now, thanks :relieved: 

and we'll be able to changed them later anyway.